openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,428 @@
+        return date_originally_created, last_modified_at_source
+
+    def extract_creator_info(self, media_info):
+        artist_string = self.get_ext_value(media_info, "Artist")
+
+        if not artist_string:
+            return None, None
+
+        artist_elem = html.fromstring(artist_string)
+        # We take all text to replicate what is shown on Wikimedia Commons
+        artist_text = "".join(artist_elem.xpath("//text()")).strip()
+        url_list = list(artist_elem.iterlinks())
+        artist_url = self.cleanse_url(url_list[0][2]) if url_list else None
+        return artist_text, artist_url
+
+    def extract_category_info(self, media_info):
+        categories_string = self.get_ext_value(media_info, "Categories") or ""
+
+        categories_list = categories_string.split("|")
+        return categories_list
+
+    def extract_file_type(self, media_info):
+        filetype = media_info.get("url", "").split(".")[-1]
+        return None if filetype == "" else filetype
+
+    def get_license_info(self, media_info):
+        license_url = self.get_ext_value(media_info, "LicenseUrl") or ""
+        # TODO Add public domain items
+        # if license_url == "":
+        #     license_name = _get_ext_value(media_info, "LicenseShortName") or ""
+        #     if license_name.lower() in {"public_domain", "pdm-owner"}:
+        #         pass
+
+        license_info = get_license_info(license_url=license_url.strip())
+        return license_info
+
+    def get_geo_data(self, media_data):
+        geo_properties = {
+            "latitude": "GPSLatitude",
+            "longitude": "GPSLongitude",
+            "map_datum": "GPSMapDatum",
+        }
+        geo_data = {}
+        for (key, value) in geo_properties.items():
+            key_value = media_data.get(value, {}).get("value")
+            if key_value:
+                geo_data[key] = key_value
+        return geo_data
+
+    def create_meta_data_dict(self, media_data):
+        meta_data = {}
+        global_usage_length = len(media_data.get("globalusage", []))
+        media_info = self.get_media_info_dict(media_data)
+        date_originally_created, last_modified_at_source = self.extract_date_info(
+            media_info
+        )
+        categories_list = self.extract_category_info(media_info)
+        description = self.get_ext_value(media_info, "ImageDescription")
+        if description:
+            description_text = " ".join(
+                html.fromstring(description).xpath("//text()")
+            ).strip()
+            meta_data["description"] = description_text
+        meta_data["global_usage_count"] = global_usage_length
+        meta_data["date_originally_created"] = date_originally_created
+        meta_data["last_modified_at_source"] = last_modified_at_source
+        meta_data["categories"] = categories_list
+        meta_data.update(self.get_geo_data(media_data))
+        return meta_data
+
+    def cleanse_url(self, url_string):
```

Comment:
I believe this function is a left-over from before we had the ImageStore classes. We should probably use `validate_url_string` from https://github.com/WordPress/openverse-catalog/blob/2a9647abec0c78734fdd9b22f3be47450e79bbc4/openverse_catalog/dags/common/urls.py

Reply:
Well spotted! In fact it looks like `validate_url_string` gets automatically applied to the [URLColumn](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/columns.py#L503) so I don't think we need to call it here at all. I verified this by removing the url cleaning and then re-running Wikimedia locally. I added a log on `artist_url` and verified that we're getting urls like `//commons.wikimedia.org/wiki/User...` from the API, but that they are all formatted correctly in the generated TSV and in the database.